### PR TITLE
[keymap/userspace/keyboard] Add mountain keyboard, and hsuchil keymaps and userspace for various keyboards

### PR DESCRIPTION
**Pull request by hsuchil**
Imported the configuration for the mountain keyboard from TMK and added my own keymap for it.

Also adding my keymaps for a few keyboards:

- basketweave
- moonlander
- mechwild/obe
- preonic/rev3_drop
- 1upkeyboards/sweet16/v1

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/basketweave/keymaps/hsuchil/keymap.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "basketweave.h"
```
Comment (drashna) [inline keyboards/basketweave/keymaps/hsuchil/keymap.c:18]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/mechwild/obe/keymaps/hsuchil/config.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#ifndef RGBLIGHT_ENABLE
+#define RGBLIGHT_ENABLE
+#endif
```
Comment (drashna) [inline keyboards/mechwild/obe/keymaps/hsuchil/config.h:22]: this define is usually an autogenerated define, when the feature is enabled. 
 This shouldn't be defined like this. 
```suggestion
```

File: keyboards/mechwild/obe/keymaps/hsuchil/config.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#ifdef RGBLIGHT_ENABLE
+    // Redefine number of LED
+    #ifdef RGBLED_NUM
+        #undef RGBLED_NUM
+    #endif // DEBUG
+    #define RGBLED_NUM 10
+
+    #define RGBLIGHT_LAYERS
+    #define RGBLIGHT_SLEEP
+#endif // RGBLIGHT_ENABLE
```
Comment (drashna) [inline keyboards/mechwild/obe/keymaps/hsuchil/config.h:33]:
```suggestion
// Redefine number of LED
#ifdef RGBLED_NUM
#    undef RGBLED_NUM
#endif // DEBUG
#define RGBLED_NUM 10

#define RGBLIGHT_LAYERS
#define RGBLIGHT_SLEEP
```

File: users/hsuchil/hsuchil.c [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        rgblight_suspend();
+        is_rgb_timeout = true;
+    }
+}
+#endif
+
+
+#ifdef HSUCHIL_EXTRA_HOUSKEEPING
+void extra_housekeeping_task_user(void);
+#endif
```
Comment (drashna) [inline users/hsuchil/hsuchil.c:101]: This allows you to have an empty implementation here, and to add a functional one in your keymap, or elsewhere. 
```suggestion
__attribute__((weak)) void extra_housekeeping_task_user(void) {};
```

File: users/hsuchil/hsuchil.c [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+void housekeeping_task_user(void) {
+    #ifdef RGBLIGHT_TIMEOUT
+    check_rgb_timeout();
+    #endif
+
+    #ifdef HSUCHIL_EXTRA_HOUSKEEPING
+    extra_housekeeping_task_user();
+    #endif
```
Comment (drashna) [inline users/hsuchil/hsuchil.c:110]:
```suggestion
#ifdef RGBLIGHT_TIMEOUT
    check_rgb_timeout();
#endif

    extra_housekeeping_task_user();
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by hsuchil** [comment]
@drashna thanks for the review, I've addressed the comments :)

File: keyboards/ydkb/mountain/config.h [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/* Copyright 2020
```
Comment (drashna) [inline keyboards/ydkb/mountain/config.h:1]: This should be handled as a separate PR, since it's for a full keyboard, not just a keymap. 

Please pull out these files, and add them to a separate PR.

**Review by drashna — CHANGES_REQUESTED** [review]
